### ISSUE TITLE
Issues with MethodSeeker and Dynamic DeepHit when running with PBC dataset

### ISSUE THREAD

**Issue by WenjuanW** [vanderschaarlab/temporai#71]
Hi,
I have errors when trying to run dynamic deephit with MethodSeeker with PBC dataset:
To reproduce the errors, please do the following:
```
from tempor.utils.dataloaders import PBCDataLoader
dataset = PBCDataLoader(random_state=42).load()

# Provide a custom hyperparameter space to search for each type of model.

hp_space = {
    "dynamic_deephit": [
        IntegerParams(name="n_iter", low=200, high=200),
        IntegerParams(name="batch_size", low=30, high=100),
        CategoricalParams(name="lr", choices=[1e-2, 1e-3, 1e-4]),
    ],
    "ts_xgb": [
        IntegerParams(name="n_iter", low=200, high=200),
        IntegerParams(name="batch_size", low=100, high=100),
        CategoricalParams(name="lr", choices=[1e-2, 1e-3, 1e-4]),
    ],
}

# Initialize a `MethodSeeker` and provide `override_hp_space`.
seeker = MethodSeeker(
    study_name="my_automl_study",
    task_type="time_to_event",
    estimator_names=[
        "dynamic_deephit",
        "ts_xgb",
    ],
    metric="c_index",   
    dataset=dataset,
    horizon=[1,5,9],
    return_top_k=2,
    num_iter=3,  # For the sake of speed of this example, only 3 AutoML iterations.
    tuner_type="bayesian",
    # Override hyperparameter space:
    override_hp_space=hp_space,
)

best_methods, best_scores = seeker.search()
```
The error is as follows:
![Screenshot 2023-08-15 at 15 55 43](https://github.com/vanderschaarlab/temporai/assets/25642178/004b9ec0-2a2f-44e2-b8ae-217e62f5ef44)

Thanks for the help!

Best wishes,
Wenjuan

**Comment by DrShushen** [comment]
Will be looking into this shortly, thank you for your patience!

**Comment by DrShushen** [comment]
The issue has now been fixed in the latest `main` branch of TemporAI.
Please install with
```bash
pip install git+https://github.com/vanderschaarlab/temporai.git
```
until `v0.0.3` is released.

### Cause
There was a bug in the use of `batch_size` variable in `DynamicDeepHitModel::predict_survival`, ultimately causing this error.

## Fix
Fixed in #76. The code example now works:
```python
from tempor.automl.params import CategoricalParams, IntegerParams
from tempor.automl.seeker import MethodSeeker
from tempor.utils.dataloaders import PBCDataLoader

dataset = PBCDataLoader(random_state=42).load()

# Provide a custom hyperparameter space to search for each type of model.

hp_space = {
    "dynamic_deephit": [
        IntegerParams(name="n_iter", low=200, high=200),
        IntegerParams(name="batch_size", low=30, high=100),
        CategoricalParams(name="lr", choices=[1e-2, 1e-3, 1e-4]),
    ],
    "ts_xgb": [
        IntegerParams(name="n_iter", low=200, high=200),
        IntegerParams(name="batch_size", low=100, high=100),
        CategoricalParams(name="lr", choices=[1e-2, 1e-3, 1e-4]),
    ],
}

# Initialize a `MethodSeeker` and provide `override_hp_space`.
seeker = MethodSeeker(
    study_name="my_automl_study",
    task_type="time_to_event",
    estimator_names=[
        "dynamic_deephit",
        "ts_xgb",
    ],
    metric="c_index",
    dataset=dataset,
    horizon=[1, 5, 9],
    return_top_k=2,
    num_iter=3,  # For the sake of speed of this example, only 3 AutoML iterations.
    tuner_type="bayesian",
    # Override hyperparameter space:
    override_hp_space=hp_space,
)

best_methods, best_scores = seeker.search()
```
